internal/static/metrics/wal/wal_test.go

```diff
@@ -776,6 +814,54 @@ func buildSeries(nameSlice []string) seriesList {
 	return s
 }
 
+func buildMixedTypeSeries() seriesList {
+	return seriesList{
+		{
+			name:    "float_series",
+			samples: []sample{{1, 10.0, nil, nil}, {10, 100.0, nil, nil}},
+			exemplars: []exemplar.Exemplar{
+				{Labels: labels.Labels{{Name: "foobar", Value: "barfoo"}}, Value: float64(10.0), Ts: int64(1), HasTs: true},
+				{Labels: labels.Labels{{Name: "lorem", Value: "ipsum"}}, Value: float64(100.0), Ts: int64(10), HasTs: true},
+			},
+		},
+		// From now on I put -1 into the float to be different from default 0.
+		// Since we should be reading back the histograms, we should never see
+		// the -1.
+		{
+			name:    "integer histogram",
+			samples: []sample{{2, -1, tsdbutil.GenerateTestHistogram(1), nil}, {20, -1, tsdbutil.GenerateTestHistogram(100), nil}},
+			exemplars: []exemplar.Exemplar{
+				{Labels: labels.Labels{{Name: "foobar", Value: "barfoo"}}, Value: float64(10.0), Ts: int64(2), HasTs: true},
+				{Labels: labels.Labels{{Name: "lorem", Value: "ipsum"}}, Value: float64(100.0), Ts: int64(20), HasTs: true},
+			},
+		},
+		{
+			name:    "float histogram",
```

Comment:
Is it worth testing an edge case where both float and integer histograms are set? Or is there something that will reliably prevent it from happening earlier on?

Reply:
The current WAL code doesn't much care about commit order for mixed typed series. We could write a unit test to demonstrate it, but maybe it would be better to have a separate PR and take inspiration from https://github.com/prometheus/prometheus/pull/17071.